src/execution/operator/persistent/base_csv_reader.cpp

```diff
@@ -343,6 +447,58 @@ bool BaseCSVReader::Flush(DataChunk &insert_chunk, bool try_add_line) {
 				// use the date format to cast the chunk
 				success = TryCastTimestampVector(options, parse_chunk.data[col_idx], insert_chunk.data[insert_idx],
 				                                 parse_chunk.size(), error_message);
+			} else if ((return_types[col_idx].id() == LogicalTypeId::FLOAT) ||
+			           (return_types[col_idx].id() == LogicalTypeId::DOUBLE)) {
+				auto type_id = return_types[col_idx].InternalType();
```

Comment:
Can we perhaps move these blocks to separate functions similar to the `TryCastDateVector` and `TryCastTimestampVector` functions?